driver_test.go

```diff
@@ -2938,3 +2938,60 @@ func TestValuerWithValueReceiverGivenNilValue(t *testing.T) {
 		// This test will panic on the INSERT if ConvertValue() does not check for typed nil before calling Value()
 	})
 }
+
+// TestRawBytesAreNotModified checks for a race condition that arises when a query context
+// is canceled while a user is calling rows.Scan. This is a more stringent test than the one
+// proposed in https://github.com/golang/go/issues/23519. Here we're explicitly closing the
+// context ourselves (instead of letting the query timeout) and using `sql.RawBytes` to check
+// the contents of our internal buffers. In theory, the `sql.RawBytes` should be invalidated
+// after cancellation, but we can still check them to ensure we never modify the underlying
+// storage.
+func TestRawBytesAreNotModified(t *testing.T) {
+	const blob = "0123456789abcdef"
+	const contextRaceIterations = 10
+	const blobSize = 64 * 1024
+	const insertRows = 64
```

Comment:
defaultBufSize is 4096 bytes.  64KiB seems too large to reproduce this issue.
Maybe, this issue is reproduced easily with only:

```go
const blobSize = 3000 // > defaultBufSize/2
const insertRows = 2
```

Reply:
contents of the blob should be different between rows, to detect race well without `-race` option.

Reply:
I've changed the `blobSize` to 3000, but I cannot reproduce the issue reliably unless we insert ~50 rows.